src/lparser.py

```diff
@@ -0,0 +1,273 @@
+# Defining parser methods
+
+# NOTE: language grammar rules must be written as the docstring for these methods.
+# The 'p' parameter is a python tuple, interpreted as a tree.
+
+# NOTE: p[0] represents the non-terminal on the left of the colon ":"
+# p[1->n] are the terminals/non-terminals to the right of the colon.
+# these can have multiple values specified between "or"s "|".
+# The idea is to pass up predicable tuples with predictable values inside them up to p_videout.
+
+# NOTE: will cause errors trying to build parser if grammar contains terminals/non-terminals
+# that have not yet been implemented.
+# =================================================================================================
+
+
+# =================================================================================================
+# Main parse method.
+
+# Global dictionary to hold all variables created or modified
+env = {}
+
+def p_videout(p): # Starting parser method.
+    '''
+     videout : var_assign
+             | methodcall
+             | NUMBER
+             | BOOLEAN
+             | empty
+    '''
+    # run(p[1])  # Run the parsed tree received from the parser.
+
+
+# ====================================================================================================
+# Assign Variables.
+def p_var_assign(p):
+    '''
+    This method permits the user assign something to a variable, declared as the lexer params (Init, STRING, etc)
+
+    var_assign : IDENTIFIER ASSIGN Init
+               | IDENTIFIER ASSIGN STRING
+               | IDENTIFIER ASSIGN NUMBER
+               | IDENTIFIER ASSIGN BOOLEAN
+               | IDENTIFIER ASSIGN IDENTIFIER
+
+    '''
+    env[p[1]] = p[3]
+    p[0] = p[1]
+
+
+def p_init(p):
+    '''
+
+    This method is used to create variables of said params and use them with the rest of the methods.
+
+    Init : videoInit
+         | photoInit
+         | concatenateClip
+    '''
+    p[0]= p[1]
+
+def p_videoInit(p):
+    '''
+
+    Method used to create a video clip taken from your file directory. Path must be specified
+    and all 'slashes (/) must be double for the python parser to work and obtain the file.
+
+    The user specifies the video from path and the run time. You select the start and end time of the clip
+    taken from the original video.
+
+    videoInit : VIDEO FROM STRING BETWEEN INT COMMA INT AND INT COMMA INT
+    '''
+    p[0] = videoClip(clip=p[3],start_time=(p[5],p[7]),end_time=(p[9],p[11]),fps=23.98)
+
+
+def p_photoInit(p):
+    '''
+
+    Method used to create a photo clip from a picture on directory. Path must be specified
+    and all 'slashes (/) must be double for the python parser to work and obtain the file.
+
+    The user specifies the picture from path and the run time (amount of time the picture will be on screen)
+
+    photoInit : PHOTO FROM STRING LASTING INT
+    '''
+
+    p[0] = photoClip(image=p[3], duration=p[5])
+
+def p_concatenateClip(p):

```

Comment:
This method seems to violate the idea of creating a tree that gets run in the "run" method by executing code instead of parsing it into a tree.

Reply:
This is what was fundamentally changed: a tree is not needed for the parser to successfully execute methods and it only adds difficulty and complexity to the code.
It seemed more useful and efficient to discard said tree hierarchy and parse it directly by using a global dictionary to store whatever variables needed for the user to run the methods. 

With this I reply to all the comments below.